### ISSUE TITLE
Support of transfer state in resolvers

### ISSUE THREAD

**Issue by AxeMaster99** [scullyio/scully#1349]
### Description

It seems like transfer state does not work in resolvers. That is due to the fact that the state is cleared when navigating, while filtering undefined values in the state$ observable. Therefore, the getState method will never emit events, causing the resolver to wait for a response infinitely.

in TransferStateService
```
private state$ = this.stateBS.pipe(filter((state) => state !== undefined));
```
and
```
 /** reset the state, so new components will never get stale data */
tap(() => this.stateBS.next(undefined)),
```

### Describe the solution you'd like

listen to NavigationEnd AND GuardsCheckEnd (and/or ResolveStart) events:
```
/** prevent emitting before navigation to _this_ URL is done. */
switchMap((e: NavigationStart) =>
  this.router.events.pipe(
    filter(
      (ev) => (ev instanceof GuardsCheckEnd || ev instanceof NavigationEnd) && ev.url === e.url,
    ),
    first(),
  ),
),
map((e: GuardsCheckEnd | NavigationEnd) => basePathOnly(e.urlAfterRedirects || e.url)),
```

**Comment by denisyilmaz** [comment]
would love to see this feature.

**Comment by SanderElias** [comment]
This is implemented in #1529
